src/app/gallery/checkout/page.tsx

```diff
@@ -0,0 +1,67 @@
+'use client'
+import { useEffect, useState } from 'react';
+import { Button } from '~/components/ui/button';
+import { stripePromise } from './load-stripe';
+import { Alert, AlertDescription, AlertTitle } from "~/components/ui/alert"
+
+
+
+export default function PreviewPage() {
+
+    // eslint-disable-next-line
+    const stripe = stripePromise;
+
+    const [isSuccess, setIsSuccess] = useState(false);
+    const [isCanceled, setIsCanceled] = useState(false);
+
+
+    useEffect(() => {
+        // NOTE: Check to see if this is a redirect back from Checkout.
+        // NOTE: `window` object only exists on the client-side, if it is used outside the 
+        // useEffect() would cause 'undefined', because of running on server-side rendering.
+        // So, it needs to call useState() first.
+        const query = new URLSearchParams(window.location.search);
+
+        // NOTE: Update the state.
+        const success = query.get('success') !== null;
+        const canceled = query.get('canceled') !== null;
+        setIsSuccess(success);
+        setIsCanceled(canceled);
+
+        if (success) {
```

Comment:
Delete this; this is handy for debugging, but doesn't add anything to the end user. Once you're done testing if it works, delete that code.

Reply:
I might leave it here from now, because if I delete this code, it won't restart the state.